lib/aee.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+class AsyncEventEmitter {
+  constructor() {
+    this.events = new Map();
+    this.wrappers = new Map();
+  }
+
+  on(name, fn) {
+    const event = this.events.get(name);
+    if (event) event.add(fn);
+    else this.events.set(name, new Set([fn]));
```

Comment:
Nit, this may be:
```js
let events = this.events.get(name);
if (!events) {
  events = new Set();
  this.events.set(name, events);
}
events.add(fn);
```